easybuild/easyconfigs/s/SCOTCH/SCOTCH-7.0.3-gompi-2022b.eb

```diff
@@ -0,0 +1,25 @@
+name = 'SCOTCH'
+version = '7.0.3'
+
+homepage = 'https://www.labri.fr/perso/pelegrin/scotch/'
+description = """Software package and libraries for sequential and parallel graph partitioning,
+static mapping, and sparse matrix block ordering, and sequential mesh and hypergraph partitioning."""
+
+toolchain = {'name': 'gompi', 'version': '2022b'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://gitlab.inria.fr/scotch/scotch/-/archive/v%(version)s/']
+sources = ['%(namelower)s-v%(version)s.tar.gz']
+checksums = ['5b5351f0ffd6fcae9ae7eafeccaa5a25602845b9ffd1afb104db932dd4d4f3c5']
+threadedmpi = False
```

Comment:
So, i tried following the dicussion, but i'm still hazy on this. You say we *still* need to disable this, but we have never had a SCOTCH easyconfig that set this before. As far as i can tell, this was always the default (actually, default in gompi, default on in iimpi), so should we just leave it out in this easyconfig as well?

Reply:
Indeed, we *still* need to disable this. It was *always* broken in the builds of OpenFOAM done by EB as far as I know.

Reply:
Having threaded MPI version of SCOTCH is the source of the following long-standing issues: https://github.com/easybuilders/easybuild-easyconfigs/issues/7093
https://github.com/easybuilders/easybuild-easyconfigs/issues/15907
and is discussed upstream: 
https://develop.openfoam.com/Development/openfoam/-/issues/2791

Reply:
>Indeed, we still need to disable this.

I thought we were already disabling it by default, but i misread the code;
https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/s/scotch.py#L109
I missed the `not` in `not in` 

But the comment in the easyblock suggests this is some old crap that should be removed a looong time ago.

```python
        # For backwards compatability of v2.8.0 this is necessary but could be removed on a major version upgrade
        mpi_family = self.toolchain.mpi_family()
        if self.cfg['threadedmpi'] is None and mpi_family not in [toolchain.INTELMPI, toolchain.QLOGICMPI]:
            cflags += " -DSCOTCH_PTHREAD"
```

version 2.8.0 of what i wonder.. it couldn't be SCOTCH since that would be too ancient. I traced it back to this PR:
https://github.com/easybuilders/easybuild-easyblocks/pull/914
so it was referring to EasyBuild 2.8.0

So, i guess this part of the easyblock should have been removed a long time ago.